### PR TITLE
Keep table heading visible after scrolling.

### PR DESCRIPTION
**Pull request by vdbergh**
Also: cleanups of latest scroll patch.

### CODE REVIEW

File: server/fishtest/templates/tests_view.mak [hunk 1]
```diff
@@ -10,10 +10,13 @@
 
 % if show_task >= 0:
 <script>
-  document.documentElement.style="scroll-behavior:auto !important; overflow:hidden !important";
+  document.documentElement.style="scroll-behavior:auto; overflow:hidden;";
   function scroll_to(task_id) {
-    document.getElementById("task" + task_id)
-      .scrollIntoView();
+    task_offset = document.getElementById("task" + task_id).offsetTop;
```
Comment (peregrineshahin) [inline server/fishtest/templates/tests_view.mak:15]: suggestions on little cleanup comments on the places that needs some cleanups as I didn't test the effects yet:
1- Variables should be declared as const or let.
2- We use camelCase for JavaScript and snake_case for python.

File: server/fishtest/templates/tests_view.mak [hunk 2]
```diff
@@ -341,7 +344,7 @@
 <div id="tasks"
      class="overflow-auto ${'collapse show' if tasks_shown else 'collapse'}">
   <table class='table table-striped table-sm'>
-    <thead ${'class="sticky-top"' if show_task == -1 else "" | n}>
+    <thead id="tasks_head" class="sticky-top">
```
Comment (peregrineshahin) [inline server/fishtest/templates/tests_view.mak:347]: Better use kebab-case for id's

File: server/fishtest/templates/tests_view.mak [hunk 3]
```diff
@@ -38,7 +41,7 @@
     </script>
 % endif
 
-<div id="enclosure" ${"style=\"visibility:hidden !important;\"" if show_task >= 0 else "" |n}>
+<div id="enclosure" ${"style=\"visibility:hidden;\"" if show_task >= 0 else "" |n}>
```
Comment (peregrineshahin) [inline server/fishtest/templates/tests_view.mak:44]: For consistency, I think it is better to use a combination of single quotes and double quotes,
something like:
`<div id="enclosure" ${'style="visibility:hidden;"' if show_task >= 0 else "" |n}>`
would be better read no?

**Review by peregrineshahin — COMMENTED** [review]
Mind these small suggestions.

**Comment by vdbergh** [comment]
> Mind these small suggestions.

I fixed all your comments, except for the use of camel case in Javascript. All my code in fishtest is snake_case, so I better keep using that for consistency (changing all the code would be too much work now). Besides I think snake_case is much easier on the eye.

**Comment by ppigazzini** [comment]
Mmm, the server workflow is a couple of weeks that fails randomly with:
```
======================================================================
ERROR: test_10_get_bench (test_run.CreateRunTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/fishtest/fishtest/server/tests/test_run.py", line 9, in test_10_get_bench
    self.assertTrue(re.match("[0-9]{7}|None", str(get_master_info()["bench"])))
TypeError: 'NoneType' object is not subscriptable

----------------------------------------------------------------------
```

**Comment by vdbergh** [comment]
> Mmm, the server workflow is a couple of weeks that fails randomly with:
> 
> ```
> ======================================================================
> ERROR: test_10_get_bench (test_run.CreateRunTest)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/runner/work/fishtest/fishtest/server/tests/test_run.py", line 9, in test_10_get_bench
>     self.assertTrue(re.match("[0-9]{7}|None", str(get_master_info()["bench"])))
> TypeError: 'NoneType' object is not subscriptable
> 
> ----------------------------------------------------------------------
> ```

I noticed that too. We should debug this.

**Comment by ppigazzini** [comment]
DEV updated with this and the #1500

**Comment by ppigazzini** [comment]
LGTM

**Review by ppigazzini — APPROVED** [review]

**Comment by ppigazzini** [comment]
PROD updated, thank you @vdbergh :)

**Pull request merged**